src/ziggurat/messaging/producer.clj

```diff
@@ -77,13 +77,14 @@
   (.borrowObject pool))
 
 (defn- publish-internal
-  [exchange message-payload expiration retry-counter]
+  [exchange message-payload is-payload-serialized? expiration retry-counter]
   (try
     (let [ch (borrow-from-pool cpool/channel-pool)]
       (try
-        (lb/publish ch exchange "" (nippy/freeze (dissoc message-payload :headers))
-                    (properties-for-publish expiration (:headers message-payload)))
-        :success
+        (let [serialized-payload (if is-payload-serialized? message-payload (nippy/freeze (dissoc message-payload :headers)))
```

Comment:
since `is-payload-serialized?` is coming in as argument, we should write a unit test for this logic

there's no change in producer_test.clj